packages/block-editor/src/hooks/child-layout.js

```diff
@@ -96,7 +108,7 @@ export function ChildLayoutEdit( {
 							},
 						} );
 					} }
-					value={ flexSize }
+					value={ flexSize || blockSize }
```

Comment:
Using the value as a fallback in all cases appears to cause an odd behaviour when editing / removing the fixed value altogether, because it winds up getting a closer to fresh value for the width. Here's a screengrab (initially set to `200px`, then after deleting the value and clicking out of the field, it sets it to `95px`:

![2022-11-30 13 00 34](https://user-images.githubusercontent.com/14988353/204689556-11a159d5-d76f-43f2-ba75-48dd64eac903.gif)

Would it work to use `blockSize` within the `onChange` handler when switching between `fit` and `fixed` instead?

Reply:
I'm not sure I understand the question. If we remove the fallback from `value` the field won't be pre-populated, which I think is a requirement?

Reply:
>  If we remove the fallback from value the field won't be pre-populated, which I think is a requirement?

Ah, good point, my apologies, I missed that detail of the requirement that James mentioned. I'd been thinking of the Fit -> Fixed selection and not the saved empty Fixed state problem 🤔

Reply:
No worries, always worth asking!

Reply:
Just thought I'd hack around a little bit. The only other way I could come up with addressing that requirement while preventing the `95px` issue from showing up, would be to add a bit of logic so that when the block is initially selected, it displays as if `fit` were selected if `fixed` is selected but `flexSize` is empty:

```js
	const [ showFit, setShowFit ] = useState();

	useEffect( () => {
		if ( selfStretch === 'fixed' && ! flexSize ) {
			setShowFit( true );
		}
	}, [] );
```

Then, in the `ToggleGroupControl` change handler we'd call `setShowFit( false )` whenever the value changes. And the bits where we're referring to `selfStretch` would become the slightly awkward:

```
				value={ showFit ? 'fit' : selfStretch || 'fit' }
				help={ helpText( showFit ? 'fit' : selfStretch ) }
```

With that in place, I think we could then remove the `blockSize` fallback in `value={ flexSize || blockSize }`, but overall it would be slightly different behaviour than what was asked for.

Feel free to ignore all that, though, I was mostly curious to better understand how it's working! 🙂

Reply:
Wouldn't that hide the input altogether when it has no value? I'm not sure we want that; the idea is that if 'fixed' is selected the input should be pre-populated with the current width/height of the block, so we have to pass `blockSize` as the value unless there's an existing `flexSize`.

Reply:
> Wouldn't that hide the input altogether when it has no value?

Yes, it would hide it when initially selected if it doesn't have a `flexSize` value, and would default to display 'fit' instead. I'm not really advocating for that, but it was the only behaviour I could figure out that avoided accidentally displaying the wrong width value when a user clears out the Fixed input field and then tabs away.

Reply:
Another thought: the issue of the wrong width value only appears when you tab away from the cleared input field, and is correct again if you select another block and then click to re-select the block you were initially working on. Personally, I'd be fine with the idea of merging this PR as-is, and us continue to fix/tweak in follow-ups if need be.